### PR TITLE
Create pagination component

### PR DESCRIPTION
**Pull request by teganduong**
The consuming component (from ko-search-support or etc) would pass down 2 props to the Pagination component:
- currentPage: current page of results
- totalPages: total number of pages
- handlePageChange: Pagination component would pass the pageNumber value to the consuming component through this function so that the consuming component can do whatever actions needed (setting the current page number and fetching the items for that page, etc)

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -6,6 +6,6 @@
 </head>
 <body>
   <script src="./bundle.js"></script>
-  <script> mountPlayground({"loaderUri":"./loader/index.html","projectKey":"/Users/evan/Code/kofile/react-components","webpackConfigType":"default"}); </script>
+  <script> mountPlayground({"loaderUri":"./loader/index.html","projectKey":"/Users/teganduong/Desktop/react-components","webpackConfigType":"default"}); </script>
```
Comment (neezer) [inline docs/index.html]: Hmm, not your fault, but this is gonna churn with all future PRs... wonder if I can fix this through Cosmos config (note to self).

File: packages/react-components-pagination/src/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+export default class Pagination extends React.Component {
+  static propTypes = {
+    totalPages: PropTypes.number.isRequired,
+    handlePageChange: PropTypes.func.isRequired
+  }
+
+  state = {
+    currentPage: 1
+  }
```
Comment (neezer) [inline packages/react-components-pagination/src/index.js]: We need to accept `currentPage` as an external prop. We keep track of pagination in the URL bar (outside of this component), so if the user refreshes the page, we currently won't be able to pass that value in. So the data would display the results from Page 5, but this component would show that we were on Page 1.

File: .eslintignore [hunk 3]
```diff
@@ -1,3 +1,5 @@
 flow-typed/**/*
 docs/**/*
 **/lib/**/*
+node_modules
+yarn.lock
```
Comment (neezer) [inline .eslintignore]: `yarn.lock` needs to be check-in!

File: packages/react-components-pagination/src/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    const validNext = currentPage < this.props.totalPages
+    const props = {
+      ...this.props,
+      currentPage,
+      pages: getPages(this.props.totalPages),
+      validPrev,
+      validNext,
+      goToPage: this.goToPage,
+      handlePrev: () => (validPrev ? this.goToPage(currentPage - 1) : {}),
+      handleNext: () => (validNext ? this.goToPage(currentPage + 1) : {})
```
Comment (neezer) [inline packages/react-components-pagination/src/index.js]: Hmm, also I don't see a `handleLast` or `handleFirst`... we should have the ability to jump to the beginning and the end... we might have hundreds of pages. Can we add those back in please?

File: packages/react-components-pagination/src/PageNavButton.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  </button>
+)
+
+PageNavButton.displayName = 'PageNavButton'
+
+PageNavButton.propTypes = {
+  onChange: PropTypes.func,
+  text: PropTypes.string,
+  disabled: PropTypes.bool
+}
```
Comment (neezer) [inline packages/react-components-pagination/src/PageNavButton.js]: General Note:

You ***should*** provide PropTypes on all components ***if you are not using Flow annotations***. @teganduong , you did that correctly here.

If you provide flow annotations, then ***you should only provide PropTypes on the entry component***. This is what the `OrderReceipt` is doing.

Just wanted to make a note of that difference. 😄

File: .eslintignore (hunk 3 above)
Comment (teganduong) [reply .eslintignore]: @neezer This is for .eslintignore

File: packages/react-components-pagination/src/PageNavButton.js (hunk 5 above)
Comment (neezer) [reply packages/react-components-pagination/src/PageNavButton.js]: That said, @teganduong — looks like these aren't defaulted at all. Any non-defaulted PropTypes should be marked as `isRequired`.

File: packages/react-components-pagination/src/View.js [hunk 6]
```diff
@@ -11,27 +11,39 @@ const buttonRow = css`
...
+            <span key={`page-${index}`} className={span}>
+              {value}
+            </span>
+          )
+        }
+
+        return (
+          <PageButton
+            key={`page-${index}`}
+            onChange={props.handlePageChange}
```
Comment (neezer) [inline packages/react-components-pagination/src/View.js]: Nitpick, but a `button` doesn't "change" like a form input does. I'd rename this to `onClick`.

**Pull request merged**